sdcm/provision/scylla_yaml/node_builder.py

```diff
@@ -0,0 +1,128 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+
+from typing import Optional, List
+
+import attr
+
+from sdcm.provision.common.builders import builder_property, AttributeProvider
+from sdcm.provision.scylla_yaml.auxiliaries import ScyllaYamlBuilderBase, SeedProvider
+
+
+@attr.attrs(kw_only=True, order=False, str=False, init=True)
+class ScyllaYamlNodeBuilder(ScyllaYamlBuilderBase):
+    seed_address: str = attr.attrib(default=None, metadata={'input_param': True}, eq=False)
+    private_ip_address: str = attr.attrib(default=None, metadata={'input_param': True}, eq=False)
+    public_ip_address: str = attr.attrib(default=None, metadata={'input_param': True}, eq=False)
+    ipv6_ip_address: str = attr.attrib(default=None, metadata={'input_param': True}, eq=False)
+
+    @property
+    def seed_provider(self) -> Optional[List[SeedProvider]]:
+        if self.seed_address is None:
+            return None
+        return [
+            SeedProvider(
+                class_name='org.apache.cassandra.locator.SimpleSeedProvider',
+                parameters=[{'seeds': self.seed_address}]
+            )
+        ]
+
+    @builder_property
+    def listen_address(self) -> Optional[str]:
+        if self._is_ip_ssh_connections_ipv6:
+            self._require_attribute(
+                'ipv6_ip_address', 'params["ip_ssh_connections"] == "ipv6", but no `ipv6_ip_address` is passed')
+            return self.ipv6_ip_address
+        if self.params.get('extra_network_interface'):
+            # Scylla should be listening on all interfaces
+            return '0.0.0.0'
+        self._require_attribute(
+            'private_ip_address', 'No `private_ip_address` has been passed')
+        return self.private_ip_address
+
+    @builder_property
+    def rpc_address(self) -> Optional[str]:
+        if self._is_ip_ssh_connections_ipv6:
+            self._require_attribute(
+                'ipv6_ip_address', 'params["ip_ssh_connections"] == "ipv6", but no `ipv6_ip_address` is passed')
+            return self.ipv6_ip_address
+        if self.params.get('extra_network_interface'):
+            # Scylla should be listening on all interfaces
+            return '0.0.0.0'
+        self._require_attribute(
+            'private_ip_address', 'No `private_ip_address` has been passed')
+        return self.private_ip_address
```

Comment:
It looks like complete duplication of lines 43-52. Please, reuse code.

Reply:
fixed